runtime/gc_glue_java/ObjectModelDelegate.hpp

```diff
@@ -373,9 +411,10 @@ class GC_ObjectModelDelegate
 	 * Constructor receives a copy of OMR's object flags mask, normalized to low order byte. Delegate
 	 * realigns it for internal use.
 	 */
-	GC_ObjectModelDelegate(fomrobject_t omrHeaderSlotFlagsMask)
+	GC_ObjectModelDelegate(uintptr_t omrHeaderSlotFlagsMask)
```

Comment:
Does this change require to remove cast in `GC_ObjectModelBase()` constructor?
```
	GC_ObjectModelBase()
		: _delegate((fomrobject_t)OMR_OBJECT_METADATA_FLAGS_MASK)
	{
```

Reply:
I don't think so, as fomrobject_t will always be castable to uintptr_t (it will always be smaller or equal to full pointer). I'm happy to change the cast to uintptr_t for consistency. Based on the comment, the field should really be uint8_t.